pipeline/full_pipeline.py

```diff
@@ -0,0 +1,230 @@
+# Copyright 2021, Blue Brain Project, EPFL
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Script that run the full pipeline."""
+from __future__ import annotations
+
+import argparse
+import logging
+import sys
+from pathlib import Path
+
+from atldld.utils import get_experiment_list_from_gene
+
+logger = logging.getLogger("full-pipeline")
+
+
+def parse_args():
+    """Parse command line arguments.
+
+    Returns
+    -------
+    args : argparse.Namespace
+        The parsed command line arguments.
+    """
+    """Parse arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--nissl-path",
+        type=Path,
+        help="""\
+        Path to Nissl Volume.
+        """,
+    )
+    parser.add_argument(
+        "--ccfv2-path",
+        type=Path,
+        help="""\
+        Path to CCFv2 Volume.
+        """,
+    )
+    parser.add_argument(
+        "--ccfv3-path",
+        type=Path,
+        help="""\
+        Path to CCFv3 Volume.
+        """,
+    )
+    parser.add_argument(
+        "--gene-name",
+        type=Path,
+        help="""\
+        Gene Expression to use.
+        """,
+    )
+    parser.add_argument(
+        "--interpolator-name",
+        type=str,
+        choices=("rife", "cain", "maskflownet", "raftnet"),
+        help="""\
+        Name of the interpolator model.
+        """,
+    )
+    parser.add_argument(
+        "--interpolator-checkpoint",
+        type=str,
+        help="""\
+        Path of the interpolator checkpoints.
+        """,
+    )
+    parser.add_argument(
+        "--reference-path",
+        type=Path,
+        help="""\
+        Path to the reference volume used for the optical flow prediction.
+        If interpolation model is "cain" or "rife", there is no need to 
+        specify a reference path.
+        """,
+    )
+    parser.add_argument(
+        "--output-dir",
+        type=Path,
+        help="""\
+        Path to directory where to save the results.
+        """,
+    )
+    parser.add_argument(
+        "-f",
+        "--force",
+        action="store_true",
+        help="""\
+        If True, force to recompute every steps.
+        """,
+    )
+    return parser.parse_args()
```

Comment:
Again, I would start using positionals. However, if there are too many of them feel free to go for options and set `required=True`

Reply:
What do you think about putting stuff related to interpolation optional though ?

Reply:
Some of the models don't require checkpoints, right?

Reply:
Yes, but only one (the linear one) and I am not sure we are going to use it often. But yes there is a model without any checkpoints. 
I am wondering if we should put a default value for `interpolation_model` and `interpolation_checkpoint` and if yes, which value we should put for the second one.